baseline-refaster-rules/src/main/java/com/palantir/baseline/refaster/AssertjOptionalHasValue.java

```diff
@@ -36,6 +36,12 @@ void before2(Optional<T> optional, T innerValue) {
         assertThat(optional.isPresent() && optional.get().equals(innerValue)).isTrue();
     }
 
+    @BeforeTemplate
+    void redundantAssertion(Optional<T> optional, T innerValue) {
+        assertThat(optional).isPresent();
+        assertThat(optional).hasValue(innerValue);
```

Comment:
I realised that lots of different cases will actually reduce down to this form eventually (using other refaster rules), so we just need this one reduction.